slick/src/main/scala/slick/jdbc/PostgresProfile.scala

```diff
@@ -209,15 +209,13 @@ trait PostgresProfile extends JdbcProfile {
 
   class UpsertBuilder(ins: Insert) extends super.UpsertBuilder(ins) {
     override def buildInsert: InsertBuilderResult = {
-      val update = "update " + tableName + " set " + softNames.map(n => s"$n=?").mkString(",") + " where " + pkNames.map(n => s"$n=?").mkString(" and ")
-      val nonAutoIncNames = nonAutoIncSyms.map(fs => quoteIdentifier(fs.name)).mkString(",")
-      val nonAutoIncVars = nonAutoIncSyms.map(_ => "?").mkString(",")
-      val cond = pkNames.map(n => s"$n=?").mkString(" and ")
-      val insert = s"insert into $tableName ($nonAutoIncNames) select $nonAutoIncVars where not exists (select 1 from $tableName where $cond)"
-      new InsertBuilderResult(table, s"$update; $insert", ConstArray.from(softSyms ++ pkSyms))
+      val insert = s"""insert into $tableName (${allNames.mkString(",")}) values $allVars"""
```

Comment:
You should provide a UUID-based random table name alias for the table here

```
val tmpName = s"$tableName-${UUID.randomUUID}"
val insert = s"""insert into $tableName as $tmpName ...
val update = s"""update set ${softNames.map(n => s"$n=$tmpName.$n").mkString(",")}"""
```

Otherwise the first user to have a table named EXCLUDED will run into problems.

Reply:
Got it, thanks